### DISCUSSION TITLE
Root tasks not executing

### DISCUSSION THREAD

**Discussion by moltar** [vercel/turborepo#7543] [Help]
### Summary

Not sure if I am "holding it wrong", as I am relatively new to turbo.

I have defined root tasks according to "[Running tasks from the root](https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#running-tasks-from-the-root)" docs.

And I can run them explicitly, as such:

```sh
turbo run compile --filter=//
```

I get expected output:

```
//:compile: cache hit, replaying logs 32e763304de77b23
//:compile: 
//:compile: 
//:compile: > @foo/bar-root@0.0.0 compile /.../dir-name
//:compile: > compile
//:compile: 
//:compile: 👾 compile | tsc --build
```

But when I run without a filter, the root task does not execute, even when there's a cache miss from the upper topo package.

### Additional information

I have a bit of an odd-ball scenario here, which might be an edge case.

I am developing packages _for_ the monorepo tooling, and then dog-fooding these packages immediately, in the workspace root.

Thus, I am installing workspace packages into the workspace root package.

**Comment by anthonyshew** [comment]
Please provide a reproduction repo and we'll see if we can help out.
